source/ShipInfoPanel.cpp

```diff
@@ -463,32 +473,93 @@ void ShipInfoPanel::DrawWeapons(const Rectangle &bounds)
 	Color topColor;
 	bool hasTop = false;
 	auto layout = Layout(static_cast<int>(LABEL_WIDTH), Truncate::BACK);
+
+	// First Calculate how many pages are needed. One is always there.
+	pages = 1;
+	int gunIndex = 0;
+	int turretIndex = 0;
+	int rowsPerPage = static_cast<int>((sprite->Height() * scale) / 10.) - 1;
+	int offset = 0;
+	while(pageNeeded)
```

Comment:
The page calculation can be moved to `UpdateInfo`, because we only need to calculate it once for any given ship.

Reply:
ok

Reply:
took a look, wont work because I need the scale and the sprite to calculate this stuff